### PR TITLE
Add gsp functions: SetAxiConfigQoSMode, UnregisterInterruptRelayQueue

### PR DESCRIPTION
**Pull request by mailwl**

### CODE REVIEW

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-02-21T12:43:54Z: https://github.com/citra-emu/citra/compare/0d086616d1af239b8e41b246bea3c1ef85fc907f...mailwl:a080fad8944333623c060951e4b339fbe8dd1a38

File: src/core/hle/service/gsp_gpu.cpp [hunk 1]
```diff
@@ -302,11 +318,17 @@ static void RegisterInterruptRelayQueue(Service::Interface* self) {
...
+    using Kernel::MemoryPermission;
+    g_shared_memory = Kernel::SharedMemory::Create(0x1000, MemoryPermission::ReadWrite,
+        MemoryPermission::ReadWrite, "GSPSharedMem");
+
     Handle shmem_handle = Kernel::g_handle_table.Create(g_shared_memory).MoveFrom();
 
     // This specific code is required for a successful initialization, rather than 0
     cmd_buff[1] = ResultCode((ErrorDescription)519, ErrorModule::GX,
-                             ErrorSummary::Success, ErrorLevel::Success).raw;
+        ErrorSummary::Success, ErrorLevel::Success).raw;
```
Comment (lioncash) [inline src/core/hle/service/gsp_gpu.cpp]: unnecessary change

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-02-21T13:55:11Z: https://github.com/citra-emu/citra/compare/0d086616d1af239b8e41b246bea3c1ef85fc907f...mailwl:0b3dfa66782a9de737ca695ffeaae77c5141d852

File: src/core/hle/service/gsp_gpu.cpp [hunk 2]
```diff
@@ -286,6 +286,22 @@ static void FlushDataCache(Service::Interface* self) {
...
+*  Outputs:
+*      1 : Result of function, 0 on success, otherwise error code
+*/
+static void SetAxiConfigQoSMode(Service::Interface* self) {
+    u32* cmd_buff = Kernel::GetCommandBuffer();
+    u32 mode = cmd_buff[1];
+
+    cmd_buff[1] = RESULT_SUCCESS.raw; // No error
+
+    LOG_DEBUG(Service_GSP, "(STUBBED) called mode=0x%08X", mode);
```
Comment (bunnei) [inline src/core/hle/service/gsp_gpu.cpp]: `LOG_WARNING`

File: src/core/hle/service/gsp_gpu.cpp [hunk 3]
```diff
@@ -302,6 +318,12 @@ static void RegisterInterruptRelayQueue(Service::Interface* self) {
     g_interrupt_event = Kernel::g_handle_table.Get<Kernel::Event>(cmd_buff[3]);
     ASSERT_MSG((g_interrupt_event != nullptr), "handle is not valid!");
 
+    g_interrupt_event->name = "GSP_Gpu::interrupt_event";
```
Comment (bunnei) [inline src/core/hle/service/gsp_gpu.cpp]: GSP_GPU

File: src/core/hle/service/gsp_gpu.cpp [hunk 4]
```diff
@@ -286,6 +286,22 @@ static void FlushDataCache(Service::Interface* self) {
 }
 
 /**
+* GSP_GPU::SetAxiConfigQoSMode service function
```
Comment (bunnei) [inline src/core/hle/service/gsp_gpu.cpp]: Spacing

File: src/core/hle/service/gsp_gpu.cpp [hunk 5]
```diff
@@ -314,6 +336,22 @@ static void RegisterInterruptRelayQueue(Service::Interface* self) {
 }
 
 /**
+* GSP_GPU::UnregisterInterruptRelayQueue service function
```
Comment (bunnei) [inline src/core/hle/service/gsp_gpu.cpp]: Spacing

**Comment by bunnei** [comment]
Feel free to squash down to one commit once you've addressed my comments. Thanks.

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-02-26T16:40:43Z: https://github.com/citra-emu/citra/compare/2f56125f12ed2a9a18226f968b44d0e0a36bdd91...mailwl:3a88809cf68daf008273196d07d3c9eb2c3e35ef

File: src/core/hle/service/gsp_gpu.cpp [hunk 6]
```diff
@@ -314,6 +336,22 @@ static void RegisterInterruptRelayQueue(Service::Interface* self) {
 }
 
 /**
+ * GSP_GPU::UnregisterInterruptRelayQueue service function
+ *  Outputs:
+ *      1 : Result of function, 0x2A07 on success, otherwise error code
```
Comment (bunnei) [inline src/core/hle/service/gsp_gpu.cpp]: What? This seems to contradict what you are returning.

**Comment by bunnei** [comment]
@mailwl do you plan on finishing this?

**Comment by mailwl** [comment]
done (was copy/paste mistake)

**Comment by bunnei** [comment]
LGTM
